waku/waku_relay/protocol.nim

```diff
@@ -214,20 +214,21 @@ proc generateOrderedValidator*(w: WakuRelay): auto {.gcsafe.} =
     return ValidationResult.Accept
   return wrappedValidator
 
-proc subscribe*(w: WakuRelay, pubsubTopic: PubsubTopic, handler: WakuRelayHandler) =
+proc subscribe*(w: WakuRelay, pubsubTopic: PubsubTopic, handler: WakuRelayHandler): TopicHandler =
   debug "subscribe", pubsubTopic=pubsubTopic
 
   # we need to wrap the handler since gossipsub doesnt understand WakuMessage
-  let wrappedHandler = proc(pubsubTopic: string, data: seq[byte]): Future[void] {.gcsafe, raises: [].} =
-    let decMsg = WakuMessage.decode(data)
-    if decMsg.isErr():
-      # fine if triggerSelf enabled, since validators are bypassed
-      error "failed to decode WakuMessage, validator passed a wrong message", error = decMsg.error
-      let fut = newFuture[void]()
-      fut.complete()
-      return fut
-    else:
-      return handler(pubsubTopic, decMsg.get())
+  let wrappedHandler =
+    proc(pubsubTopic: string, data: seq[byte]): Future[void] {.gcsafe, raises: [].} =
+      let decMsg = WakuMessage.decode(data)
+      if decMsg.isErr():
+        # fine if triggerSelf enabled, since validators are bypassed
+        error "failed to decode WakuMessage, validator passed a wrong message", error = decMsg.error
+        let fut = newFuture[void]()
+        fut.complete()
+        return fut
```

Comment:
doesn't compile since `TopicHandler` has a specific signature that does not include it.

I'm not sure, what is the purpose of these 3 lines exactly?

Reply:
> doesn't compile since `TopicHandler` has a specific signature that does not include it...

Okay. Those lines are simply to create a future and return it as completed so that the caller `await` resumes immediately.

Reply:
So it saves one yield to the async run-time?

Reply:
I'm going to leave it there since I'm not sure how it works exactly.